nextcloud: update apps, move notify_push app into `nextcloud-notify_push` package

Ma27 opened the pull request:
The latter is the result of the issues from https://github.com/NixOS/nixpkgs/pull/374902#issuecomment-2601759264.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ma27 committed:
nextcloudApps: update

SuperSandro2000 commented on pkgs/servers/nextcloud/notify_push.nix:
```diff
@@ -3,10 +3,14 @@
 rustPlatform.buildRustPackage rec {
   pname = "notify_push";
+
+  # NOTE: make sure this is compatible with all Nextcloud versions
+  # in nixpkgs!
```
Maybe we can link https://github.com/nextcloud/notify_push/blob/main/appinfo/info.xml#L30 ?

They seem to support a very wide range so hopefully this doesn't become a problem.

SuperSandro2000 reviewed (approved):
fine for me :)

dotlambda commented on pkgs/servers/nextcloud/packages/29.json:
```diff
@@ -90,9 +90,9 @@
-    "url": "https://github.com/nextcloud-releases/forms/releases/download/v4.3.5/forms-v4.3.5.tar.gz",
-    "version": "4.3.5",
+    "hash": "sha256-Fso4vxcn+4lCuWpgPMmp4dUMMeDaZrXne3CCnt4p7Rg=",
+    "url": "https://github.com/nextcloud-releases/forms/releases/download/v4.3.6/forms-v4.3.6.tar.gz",
```
This change is in the wrong commit.

Ma27 replied:
What do you mean with wrong here specifically?

dotlambda replied:
It's part of the commit moving the notify_push app rather than the one updating all apps.

dotlambda commented on pkgs/servers/nextcloud/notify_push.nix:
```diff
@@ -20,6 +26,20 @@ rustPlatform.buildRustPackage rec {
+      description = ''
+        Push update support for desktop app.
+
+        Once the app is installed, the push binary needs to be setup. You can either use the setup wizard with `occ notify_push:setup` or see the [README](http://github.com/nextcloud/notify_push) for detailed setup instructions
```
```suggestion
```

Ma27 replied:
Added a different sentence about the server-side requirements.

dotlambda commented on pkgs/servers/nextcloud/notify_push.nix:
```diff
@@ -20,6 +26,20 @@ rustPlatform.buildRustPackage rec {
+      homepage = "https://github.com/nextcloud/notify_push";
+      url = "https://github.com/nextcloud-releases/notify_push/releases/download/v${version}/notify_push-v${version}.tar.gz";
+      description = ''
+        Push update support for desktop app.
```
```suggestion
        Push update support for desktop app
```

dotlambda commented on pkgs/servers/nextcloud/notify_push.nix:
```diff
@@ -20,6 +26,20 @@ rustPlatform.buildRustPackage rec {
+      description = ''
+        Push update support for desktop app.
+        Requires a server-side notify_push installation. On NixOS, this can be done
+        with the `services.nextcloud.notify_push` module.
```
This can go in `longDescription` if you want.

Ma27 committed:
nextcloud-notify_push.app: move out of automatically generated set

`pkgs.nextcloud-notify_push` and the app
(`nextcloudPackages.apps.notify_push`) must always have the same
version. However, the latter is updated automatically. Most recently
this happened in #374902.

With this change, the app isn't updated automatically anymore, but is
part of the server-side package (and uses the same version).

dotlambda commented:
@ofborg test nextcloud
